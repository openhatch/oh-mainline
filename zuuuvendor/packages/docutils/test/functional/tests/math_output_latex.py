# Source and destination file names.
test_source = "data/math.txt"
test_destination = "math_output_latex.html"

# Keyword parameters passed to publish_file.
reader_name = "standalone"
parser_name = "rst"
writer_name = "html"

# Settings
settings_overrides['math_output'] = 'latex'
# local copy of default stylesheet:
settings_overrides['stylesheet_path'] = ( 
    'functional/input/data/html4css1.css')


