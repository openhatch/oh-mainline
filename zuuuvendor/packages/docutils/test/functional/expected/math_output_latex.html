<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.11: http://docutils.sourceforge.net/" />
<title>Mathematics</title>
<link rel="stylesheet" href="../input/data/html4css1.css" type="text/css" />
</head>
<body>
<div class="document" id="mathematics">
<h1 class="title">Mathematics</h1>

<p>Docutils supports inline math with the prefix or postfix <tt class="docutils literal">:math:</tt>
role specificator, <tt class="math">n! + \sin(x_n^2)</tt>
 and <tt class="math">A_\text{c} =
\frac{\pi}{4} d^2</tt>
, as well as displayed math via the
<cite>math</cite> directive:</p>
<pre class="math">
f(\epsilon) = \frac{1}{1 + \exp\left(\frac{\varepsilon}{k_\text{B}T}\right)}
</pre>
<p>Content may start on the first line of the directive, e.g.</p>
<pre class="math">
N = \frac{\text{number of apples}}{7}
</pre>
<p>Equations can be labeled with a reference name using the <tt class="docutils literal">:name:</tt> option.
See <a class="reference internal" href="#eq-m">eq:M</a> and <a class="reference internal" href="#eq-schrodinger">eq:schrödinger</a> below.</p>
<p>The determinant of the matrix</p>
<pre class="math" id="eq-m">
\mathbf{M} = \left(\begin{matrix}a&amp;b\\c&amp;d\end{matrix}\right)
</pre>
<p>is <tt class="math">|\mathbf{M}| = ad - bc</tt>
.</p>
<p>More than one display math block can be put in one math directive.
For example, the following sum and integral with limits:</p>
<pre class="math">
\int_0^1 x^n dx = \frac{1}{n + 1}
</pre>
<pre class="math">
\sum_{n=1}^m n = \frac{m(m+1)}{2}
</pre>
<p>LaTeX-supported Unicode math symbols can be used in math roles and
directives:</p>
<p>The Schrödinger equation</p>
<pre class="math" id="eq-schrodinger">
i\hbar \frac{\partial }{\partial t}\Psi  = \hat{H}\Psi ,
</pre>
<p>with the <em>wave function</em> <tt class="math">\Psi </tt>
, describes how the quantum state of a
physical system changes in time.</p>
<dl class="docutils">
<dt>Math-Accents:</dt>
<dd><table border="1" class="borderless first last docutils">
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<tbody valign="top">
<tr><td><tt class="math">\acute{a}</tt>
      <tt class="docutils literal">\acute{a}</tt></td>
<td><tt class="math">\dot{t}</tt>
        <tt class="docutils literal">\dot{t}</tt></td>
<td><tt class="math">\hat{\gamma}</tt>
   <tt class="docutils literal"><span class="pre">\hat{\gamma}</span></tt></td>
</tr>
<tr><td><tt class="math">\grave{a}</tt>
      <tt class="docutils literal">\grave{a}</tt></td>
<td><tt class="math">\ddot{t}</tt>
       <tt class="docutils literal">\ddot{t}</tt></td>
<td><tt class="math">\tilde{\alpha}</tt>
 <tt class="docutils literal"><span class="pre">\tilde{\alpha}</span></tt></td>
</tr>
<tr><td><tt class="math">\breve{x}</tt>
      <tt class="docutils literal">\breve{x}</tt></td>
<td><tt class="math">\dddot{t}</tt>
      <tt class="docutils literal">\dddot{t}</tt></td>
<td><tt class="math">\vec{\imath}</tt>
   <tt class="docutils literal"><span class="pre">\vec{\imath}</span></tt></td>
</tr>
<tr><td><tt class="math">\check{a}</tt>
      <tt class="docutils literal">\check{a}</tt></td>
<td><tt class="math">\bar{a}</tt>
        <tt class="docutils literal">\bar{a}</tt></td>
<td><tt class="math">\vec{R}</tt>
        <tt class="docutils literal">\vec{R}</tt></td>
</tr>
</tbody>
</table>
</dd>
</dl>
<!-- \widetilde{xxx}
\widehat{xxx} -->
<p>Modulation Transfer Function:</p>
<pre class="math">
\text{MTF} = \left|\frac{\mathcal{F}\{s(x)\}}
                        {\mathcal{F}\{ s(x)\} |_{\omega _{x}=0}}\right|
           = \mathrm{abs}\left(\frac
                 {\int _{-\infty }^{\infty }s(x) \mathrm{e}^{\mathrm{i}\omega _{x}x}\mathrm{d}{x}}
                 {\int _{-\infty }^{\infty }s(x)\mathrm{d}{x}}
             \right).
</pre>
<p>Math split over two lines: If a double backslash is detected outside a
<tt class="docutils literal"><span class="pre">\begin{...}</span> <span class="pre">\end{...}</span></tt> pair, the math code is wrapped in an <a class="reference external" href="ftp://ftp.ams.org/ams/doc/amsmath/short-math-guide.pdf">AMSmath</a>
<tt class="docutils literal">align</tt> environment:</p>
<pre class="math">
s_{\mathrm{out}}(x) &amp; = s_{\mathrm{in}}(x') * s_\delta (x-x') \\
                    &amp; = \int  s_{\mathrm{in}}(x')s_\delta (x-x')\mathrm{d}x'
</pre>
<p>Cases (&quot;manually&quot;, with <tt class="docutils literal">matrix</tt> environment):</p>
<pre class="math">
\mathrm{sgn}(x) = \left\{\begin{matrix}
                    -1 &amp; x&lt;0\\
                    1 &amp; x&gt;0
                  \end{matrix}\right.
</pre>
<p>Cases with the <a class="reference external" href="ftp://ftp.ams.org/ams/doc/amsmath/short-math-guide.pdf">AMSmath</a> <tt class="docutils literal">cases</tt> environment (not (yet) supported by
HTML writers with <tt class="docutils literal"><span class="pre">--math-output=MathML</span></tt>):</p>
<pre class="math">
\mathrm{sgn}(x) = \begin{cases}
                    -1 &amp; x&lt;0\\
                     1 &amp; x&gt;0
                  \end{cases}
</pre>
</div>
</body>
</html>
